netbox/dcim/models/racks.py

```diff
@@ -269,8 +287,8 @@ def units(self):
         Return a list of unit numbers, top to bottom.
         """
         if self.desc_units:
-            return drange(decimal.Decimal(1.0), self.u_height + 1, 0.5)
-        return drange(self.u_height + decimal.Decimal(0.5), 0.5, -0.5)
+            return drange(decimal.Decimal(self.starting_unit), self.u_height + 1, 0.5)
```

Comment:
This fixes the unit list for racks with descending unit numbering.

```suggestion
            return drange(decimal.Decimal(self.starting_unit), self.u_height + self.starting_unit, 0.5)
```